### ISSUE TITLE
Modify rule S5332: False negative when addListener is called on LoadBalancer instances (AWS CDK)

### ISSUE THREAD

**Issue by gaetan-ferry-sonarsource** [SonarSource/SonarJS#3628]
# What

Implementation ticket https://github.com/SonarSource/SonarJS/issues/3458 specifies the detection logic for rule S5332 for AWS CDK elastic load balancers types of resources. In particular, this snippet specifies the expected behavior when encountering a call to `LoadBalancer.addListener`:

> Fully qualified class name: aws-cdk-lib.aws_elasticloadbalancing.LoadBalancer
    Initializer [LoadBalancer(scope: Construct, id: string, props: LoadBalancerProps)](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_elasticloadbalancing.LoadBalancer.html#initializer)
>    *  Method: [addListener](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_elasticloadbalancing.LoadBalancer.html#addwbrlistenerlistener)
>    *  Raise an issue if:
>         *  externalProtocol parameter set to [LoadBalancingProtocol.TCP](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_elasticloadbalancing.LoadBalancingProtocol.html) or [LoadBalancingProtocol.HTTP](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_elasticloadbalancing.LoadBalancingProtocol.html).

However, a false negative was identified in this case. For example, the following piece of code does not raise any issues:

```typescript
lb = new LoadBalancer(this, 'elb-multi-listener', {
  vpc,
  internetFacing: true,
  healthCheck: {
    port: 80,
  }
});

lb.addTarget(autoScalingGroup);
lb.addListener({
  externalPort:10000,
  externalProtocol:LoadBalancingProtocol.TCP, // Sensitive
  internalPort:10000
});
lb.addListener({
  externalPort:10000,
  externalProtocol:LoadBalancingProtocol.HTTP, // Sensitive
  internalPort:10000
});
```

# How

Implement the expected logic when a bad `ListenerProps` is passed to the addListenerMethod.

**Comment by Wohops** [comment]
Migrated to Jira. New ticket number: JS-105
